thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/client/TimeRangeUtils.java

```diff
@@ -60,6 +62,42 @@ public static DateTime increment(DateTime input, TimeGranularity granularity) {
     return output;
   }
 
+  /**
+   * Given time granularity and start time (with correct time zone information), returns the bucket
+   * index of the current time (with correct time zone information).
+   *
+   * The reason to use this method to calculate the bucket index is to align the shifted data point
+   * due to daylight saving time to the correct bucket. Note that this method have no effect if
+   * the input time use UTC timezone.
+   *
+   * For instance, considering March 13th 2016, when DST takes effect. When we processing daily
+   * data, whose timestamp is aligned at 0 am at each day. The data point on March 14th would be
+   * actually aligned to 13th's bucket because the two data point only has 23 hours difference.
+   * Therefore, we cannot calculate the bucket index simply divide the timestamp by millis of 24
+   * hours.
+   *
+   * We don't check for granularity finer than DAYS because those granularity has lose precision on
+   * timestamp and the alignment doesn't help much in comparison with the default case.
+   *
+   * @param granularity the time granularity of the bucket
+   * @param start the start time of the first bucket
+   * @param current the current time
+   * @return the bucket index of current time
+   */
+  public static int computeBucketIndex(TimeGranularity granularity, DateTime start, DateTime current) {
+    int index = -1;
+    switch (granularity.getUnit()) {
+    case DAYS:
+      Days d = Days.daysBetween(start, current);
```

Comment:
It will return 0, instead of the actual date difference. You need to change to the local date fist. 
Check: http://stackoverflow.com/questions/3802893/number-of-days-between-two-dates-in-joda-time

Also, in current version, do you consider the effect on HOURS?